File: ToDoBackend/Core/Models/ToDo.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ToDoBackend.Core.Models
{
    [Table("ToDos")]
    public class ToDo
    {        
        public int Id { get; set; }

        [Required]
        [MaxLength]
        public string Name { get; set; }

        public bool Done { get; set; }

        public int PriorityId { get; set; }

        public Priority Priority { get; set; }
    }
}

File: ToDoFrontend/src/models/ToDo.ts
export interface KeyValuePair { 
    id: number; 
    name: string; 
  }
  
  export interface ToDo {
    id: number; 
    priority: KeyValuePair;
    name: string;
    done: boolean;
  }
  
  export interface SaveToDo {
    id: number; 
    priorityId: number;
    name: string;
    done: boolean;
  }